### ISSUE TITLE
Change license from MIT to `GPL-3.0 OR MulanPubL`

### ISSUE THREAD

**Issue by zjp-CN** [os-checker/os-checker#311]
NOTE: MulanPubL hasn't been added to SPDX license list yet

Tracker: https://github.com/spdx/license-list-XML/issues/2618

So currently we have to  hack by adding only GPL-3.0 as license value, and  MulanPubL license file path as license-file value in Cargo.toml. Inconsistent and confusing!

![Image](https://github.com/user-attachments/assets/cf3a4400-8c2f-49e5-91de-ae228d3d97b2)

This is warned against by `cargo publish`:

```bash
warning: only one of `license` or `license-file` is necessary
`license` should be used if the package license can be expressed with a standard SPDX expression.
`license-file` should be used if the package uses a non-standard license.
See https://doc.rust-lang.org/cargo/reference/manifest.html#the-license-and-license-file-fields for more information.
```

Also lately there are some updates on license vs license-file discussions:
* https://github.com/rust-lang/cargo/issues/5933
* https://github.com/rust-lang/cargo/issues/13328#issuecomment-2714727911
* https://github.com/rust-lang/cargo/issues/12053
* https://github.com/rust-lang/cargo/issues/8537#issuecomment-924329183
